### PR TITLE
[fakeintake] ♻️ move and refactor fakeintake

### PR DESCRIPTION
**Pull request by pducolin**
### What does this PR do?

* Move `fakeintake` from `fake-intake` to `fakeintake`
* Move server related code to `server`
* Move `api` code (response structure) to `api`
* Run tests on the CI
* Update readme after https://github.com/DataDog/datadog-agent/pull/15516/files#diff-01b7d368418dd5c99b2de1f93f79b58e7548c4f31875f497a525a0015924e7beR15

### Motivation

Preparatory refactoring to add `client` ad parsers, that can be used in integration tests

### Reviewer's Checklist

- [ ] If known, an appropriate milestone has been selected; otherwise the `Triage` milestone is set.
- [ ] Use the `major_change` label if your change either has a major impact on the code base, is impacting multiple teams or is changing important well-established internals of the Agent. This label will be use during QA to make sure each team pay extra attention to the changed behavior. For any customer facing change use a releasenote.
- [ ] A [release note](https://github.com/DataDog/datadog-agent/blob/main/docs/dev/contributing.md#reno) has been added or the `changelog/no-changelog` label has been applied.
- [ ] Changed code has automated tests for its functionality.
- [ ] Adequate QA/testing plan information is provided if the `qa/skip-qa` label is not applied.
- [ ] At least one `team/..` label has been applied, indicating the team(s) that should QA this change.
- [ ] If applicable, docs team has been notified or [an issue has been opened on the documentation repo](https://github.com/DataDog/documentation/issues/new).
- [ ] If applicable, the `need-change/operator` and `need-change/helm` labels have been applied.
- [ ] If applicable, the `k8s/<min-version>` label, indicating the lowest Kubernetes version compatible with this feature. 
- [ ] If applicable, the [config template](https://github.com/DataDog/datadog-agent/blob/main/pkg/config/config_template.yaml) has been updated.

### CODE REVIEW

**Review by ogaca-dd — APPROVED** [review]
LGTM

File: test/fakeintake/README.md
Comment (vboulineau) [inline test/fakeintake/README.md:7]: Any reason to downgrade?

File: test/fakeintake/README.md
Comment (vboulineau) [inline test/fakeintake/README.md:43]: No need to build and install `go run` would do for local testing.

File: test/fakeintake/api/api.go
Comment (vboulineau) [inline test/fakeintake/api/api.go:8]: The name feels like a function name. We're mostly interested in what the type represents more than where it's used.
I think something like `RawPayloads` or `Payloads`

File: test/fakeintake/README.md
Comment (vboulineau) [reply test/fakeintake/README.md:7]: It seems that `go.mod` was not updated but `datadog-agent` is now built with `1.19`

File: test/fakeintake/api/api.go
Comment (pducolin) [reply test/fakeintake/api/api.go:8]: I wanted to pass the info that this is the model of a `GET` response. Maybe `APIFakeIntakePayloadsGETResponse` or `APIPayloadsGETResponse` ? It would follow the naming pattern used in frontend code

**Comment by pducolin** [comment]
`integration_tests` are not passing because I am not based on latest `main`, will merge main once all changes are approved.

**Review by ogaca-dd — APPROVED** [review]

**Review by KSerrania — APPROVED** [review]
LGTM for the `.gitignore` and `tasks/modules.py` changes. The other files aren't owned by Agent Platform after the `CODEOWNERS` update.

**Pull request merged**
